request/models.py

```diff
@@ -54,14 +54,20 @@ def get_user(self):
 
     def from_http_request(self, request, response=None, commit=True):
         # Request infomation
+        
+        transform_http_headers(request)
+
         self.method = request.method
         self.path = request.path[:255]
 
         self.is_secure = request.is_secure()
         self.is_ajax = request.is_ajax()
 
         # User infomation
-        self.ip = request.META.get('REMOTE_ADDR', '')
+        # record real ip from header if was set
+        self.ip = request.META.get('HTTP_X_REAL_IP', '')
```

Comment:
This seems like a bit of a security flaw to me. Anyone could send the `X-Real-IP` header with a request to trick logging to the wrong IP.

Example:

```shell
$ curl -H 'X-Real-IP: 8.8.8.8' http://localhost:8080
```

Where my request actually originating from localhost (`127.0.0.1`/`::1`) would be logged as from `8.8.8.8`.

I'd suggest a different approach of adding a custom middleware to your own application to switch the IPs out.

```python
class IPAddressForwardingMiddleware(object):
    def prepare_request(request):
        if 'HTTP_X_REAL_IP' not in request.META:
            raise SuspiciousOperation()

        request.META['REMOTE_ADDR'] = request.META['HTTP_X_REAL_IP']
```

Then your custom middleware will run first, changing the IP address for all other middleware and the request.

You can check the Django book for a full example: http://www.djangobook.com/en/2.0/chapter17.html#what-s-middleware